packages/core/js-sdk/src/admin/product-category.ts

```diff
@@ -0,0 +1,97 @@
+import { HttpTypes } from "@medusajs/types"
+import { Client } from "../client"
+import { ClientHeaders } from "../types"
+
+export class ProductCategory {
+  private client: Client
+  constructor(client: Client) {
+    this.client = client
+  }
+
+  async create(
+    body: HttpTypes.AdminCreateProductCategory,
+    query?: HttpTypes.AdminProductCategoryParams,
+    headers?: ClientHeaders
+  ) {
+    return this.client.fetch<HttpTypes.AdminProductCategoryResponse>(
+      `/admin/product-categories`,
+      {
+        method: "POST",
+        headers,
+        body,
+        query,
+      }
+    )
+  }
+
+  async update(
+    id: string,
+    body: HttpTypes.AdminUpdateProductCategory,
+    query?: HttpTypes.AdminProductCategoryParams,
+    headers?: ClientHeaders
+  ) {
+    return this.client.fetch<HttpTypes.AdminProductCategoryResponse>(
+      `/admin/product-categories/${id}`,
+      {
+        method: "POST",
+        headers,
+        body,
+        query,
+      }
+    )
+  }
+
+  async list(
+    query?: HttpTypes.AdminProductCategoryListParams,
+    headers?: ClientHeaders
+  ) {
+    return this.client.fetch<HttpTypes.AdminProductCategoryListResponse>(
+      `/admin/product-categories`,
+      {
+        headers,
+        query: query,
+      }
+    )
+  }
+
+  async retrieve(
+    id: string,
+    query?: HttpTypes.AdminProductCategoryParams,
+    headers?: ClientHeaders
+  ) {
+    return this.client.fetch<HttpTypes.AdminProductCategoryResponse>(
+      `/admin/product-categories/${id}`,
+      {
+        query,
+        headers,
+      }
+    )
+  }
+
+  async delete(id: string, headers?: ClientHeaders) {
+    return this.client.fetch<HttpTypes.AdminProductCategoryDeleteResponse>(
+      `/admin/product-categories/${id}`,
+      {
+        method: "DELETE",
+        headers,
+      }
+    )
+  }
+
+  async updateProducts(
```

Comment:
But it's both removing and adding products, so assign doesn't work, and set seems more like of a source of truth action as in this will now be the the products that are part of this category.

Reply:
Yeah I agree neither are great, but `updateProducts` seems more like it will update the product data, not set products on categories, so `setProducts` is a tiny bit better IMO, but I leave it up to you

Reply:
Maybe `sdk.category.modifyProducts()`?

Reply:
I'll just leave it for now, and we can come up with a proper name for this type of action for all the relevant domains